connectors/src/connectors/google_drive/index.ts

```diff
@@ -543,51 +604,119 @@ export async function setGoogleDriveConnectorPermissions(
-      driveFileId: internalIds,
-    },
-  });
+  const driveFileIds = internalIds.filter(
+    (id) => !isGoogleSheetContentNodeInternalId(id)
+  );
+  const sheetIds = internalIds
+    .filter((id) => isGoogleSheetContentNodeInternalId(id))
+    .map(getGoogleIdsFromSheetContentNodeInternalId);
+
+  if (!!sheetIds.length && viewType !== "tables") {
+    return new Err(
+      new Error(
+        `Cannot retrieve Google Sheets Content Nodes in view type "${viewType}".`
+      )
+    );
+  }
 
-  const nodes = await concurrentExecutor(
+  const folderOrFiles = driveFileIds.length
+    ? await GoogleDriveFiles.findAll({
+        where: {
+          connectorId: connectorId,
+          driveFileId: driveFileIds,
+        },
+      })
+    : [];
+  const sheets = sheetIds.length
+    ? await GoogleDriveSheet.findAll({
+        where: {
+          [Op.or]: sheetIds.map((s) => ({
+            [Op.and]: [
+              {
+                driveFileId: s.googleFileId,
+              },
+              {
+                driveSheetId: s.googleSheetId,
+              },
+            ],
+          })),
+        },
+      })
+    : [];
+
+  const folderOrFileNodes = await concurrentExecutor(
     folderOrFiles,
     async (f): Promise<ContentNode> => {
-      const type = getPermissionViewType(f);
+      const type = getPermissionViewType(f, viewType);
       let sourceUrl = null;
-      if (type === "file") {
-        sourceUrl = `https://drive.google.com/file/d/${f.driveFileId}/view`;
-      } else if (type === "folder") {
-        sourceUrl = `https://drive.google.com/drive/folders/${f.driveFileId}`;
-      } else if (type === "database") {
+
+      if (isGoogleDriveSpreadSheetFile(f)) {
         sourceUrl = `https://docs.google.com/spreadsheets/d/${f.driveFileId}/edit`;
+      } else if (isGoogleDriveFolder(f)) {
+        sourceUrl = `https://drive.google.com/drive/folders/${f.driveFileId}`;
+      } else {
+        sourceUrl = `https://drive.google.com/file/d/${f.driveFileId}/view`;
       }
+
       return {
         provider: "google_drive",
         internalId: f.driveFileId,
         parentInternalId: null,
-        type: getPermissionViewType(f),
+        type,
         title: f.name || "",
         dustDocumentId: getGoogleDriveEntityDocumentId(f),
         lastUpdatedAt: f.lastUpsertedTs?.getTime() || null,
         sourceUrl,
-        expandable:
-          (await GoogleDriveFiles.findOne({
-            attributes: ["id"],
-            where: {
-              connectorId: connectorId,
-              parentId: f.driveFileId,
-            },
-          })) !== null,
+        expandable: await isDriveObjectExpandable({
+          objectId: f.driveFileId,
+          mimeType: f.mimeType,
+          connectorId,
+          viewType,
+        }),
+        permission: "read",
+      };
+    },
+    { concurrency: 4 }
+  );
+
+  const sheetNodes = await concurrentExecutor(
+    sheets,
+    async (s): Promise<ContentNode> => {
```

Comment:
No more need for an async wrapper here, right?

Reply:
fixed 🙏